### ISSUE TITLE
Incorrect/Missing inference for slash-joining `pathlib` paths -> Missing diagnostics

### ISSUE THREAD

**Issue by Flamefire** [pylint-dev/pylint#11202]
### Bug description

```python
"doc"
from pathlib import Path

Path('/tmp').read_text()				# unspecified-encoding
Path('/tmp').parents[0].read_text()		# unspecified-encoding

(Path('/') / 'tmp').read_text()			# No diagnostic
p: Path = Path('/') / 'tmp'				# Try with type annotation
p.read_text()							# No diagnostic
```
### Command used

```shell
`pylint --enable W1514 /tmp/test.py`
```

### Pylint output

```python
************* Module test
/tmp/test.py:4:0: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
/tmp/test.py:5:0: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
```

### Expected behavior

PyLint should infer the type of `Path() / "str"` as `Path` and apply the same diagnostics as to direct instances of `Path`

Even using type-annotated variables doesn't help

### Pylint version

```shell
pylint 4.0.6
astroid 4.0.4
Python 3.12.1 (main, Feb 11 2026, 13:37:21) [GCC 13.3.0]
```

May be related to long fixed https://github.com/pylint-dev/pylint/issues/224

**Comment by adityaanikam** [comment]
Reproduced this exactly on `astroid==4.0.4`/`pylint==4.0.6` (matching the versions in the report). The `unspecified encoding` checker in `stdlib.py` is purely inference based — it just checks whether `node.func`'s inferred root module is `pathlib`/`_io`, no syntactic whitelist — so in principle it should catch any expression that infers to a `Path` method. The gap is on the astroid side: `brain_pathlib.py` only has an inference tip for `Path(...).parents[N]`, nothing for `__truediv__`. Without that, astroid falls through to inferring `with_segments`'s `type(self)(*pathsegments)` call, which it can't resolve, so `Path('/') / 'tmp'` comes back `Uninferable` and the checker's continue-on-`Uninferable` branch silently skips it.

The actual fix belongs in astroid — a small brain tip mirroring the existing `parents[N]` one, registered for `PurePath.__truediv__`/`__rtruediv__`, inferring the result as the same `Path` subclass. #224 is related in spirit (same family of Path-subclass-method inference gap) but not the same code path — it was about `with_name()` losing the `Path`-vs-`PurePath` distinction and causing a false `no-member`, closed back in 2018. I'd like to put up the astroid PR for the `__truediv__` tip if that's the right place for it — let me know if you'd rather this get re filed there directly instead of tracked from this pylint issue.
